/*
 * ChannelDescriptor — per-channel capability advertisement.
 *
 * Field numbers are from aasdk (f1x.studio, 2018) and DO NOT match the
 * APK's internal protobuf-lite descriptor (wbw.java, AA v16.1). However,
 * these numbers work on the wire because the phone's protobuf parser
 * silently discards fields with mismatched wire types, and all real
 * channel configuration is negotiated per-channel during setup — not
 * from ServiceDiscoveryResponse.
 *
 * Tested 2026-02-25:
 *   - wbw field numbers + channel_id only → phone connects, opens control
 *     channel, disconnects ("incompatible" = no video/audio capability)
 *   - wbw field numbers + aasdk sub-channel data → phone connects but
 *     never opens service channels (garbled inner message field numbers)
 *   - aasdk field numbers + aasdk sub-channel data → works perfectly
 *
 * Conclusion: wbw numbers ARE the correct v16.1 wire format, but a full
 * migration requires also fixing field numbers in every sub-channel proto
 * (AVChannel/vys, SensorChannel/wbu, InputChannel/vya, etc.). No
 * functional benefit since aasdk numbers work via graceful degradation.
 *
 * APK internal (wbw) field mapping for reference:
 *   1=channel_id, 2=channel_type, 3=sensor, 4=av, 5=input,
 *   6=av_input, 7=bluetooth, 8=wifi_projection, 9=navigation,
 *   10=phone_status, 11=notification, 12=media_browser,
 *   13=media_info, 14=audio_focus, 15=wifi, 16=car_control, 17=radio
 *
 * Original aasdk copyright (field names/structure):
 * Copyright (C) 2018 f1x.studio (Michal Szwaj), GPLv3
 */

syntax="proto2";

option optimize_for=SPEED;

import "SensorChannelData.proto";
import "AVChannelData.proto";
import "InputChannelData.proto";
import "AVInputChannelData.proto";
import "BluetoothChannelData.proto";
import "NavigationChannelData.proto";
import "RadioChannelData.proto";
import "VendorExtensionChannelData.proto";
import "MediaChannelData.proto";
import "PhoneStatusChannelData.proto";
import "NotificationChannelData.proto";
import "WifiChannelData.proto";

package oaa.proto.data;

message ChannelDescriptor
{
    required uint32 channel_id = 1;
    optional SensorChannel sensor_channel = 2;
    optional AVChannel av_channel = 3;
    optional InputChannel input_channel = 4;
    optional AVInputChannel av_input_channel = 5;
    optional BluetoothChannel bluetooth_channel = 6;
    optional RadioChannel radio_channel = 7;
    optional NavigationChannel navigation_channel = 8;
    optional MediaInfoChannel media_info_channel = 9;
    optional PhoneStatusChannel phone_status_channel = 10;
    optional bytes media_browser_channel = 11;
    optional VendorExtensionChannel vendor_extension_channel = 12;
    optional NotificationChannel notification_channel = 13;
    optional WifiChannel wifi_channel = 14;
    optional bytes car_control_channel = 15;
}
